.github/workflows/build.yaml

```diff
@@ -24,6 +24,9 @@ jobs:
       - name: Lint
         run: npm run lint
 
+      - name: TypeScript Type Check
+        run: npx tsc --noEmit
```

Comment:
Using `npx` in the pipeline might be a bit unpredictable.

How about adding a script to package.json and just call it here?

```json
{
  "scripts": {
    "type-check": "tsc --noEmit"
  }
}
```